keymapper/migrations.py

```diff
@@ -0,0 +1,133 @@
+    presets = []
+    for folder in preset_path.iterdir():
+        if not folder.is_dir():
+            continue
+
+        for preset in folder.iterdir():
+            if preset.suffix == ".json":
+                presets.append(preset)
+    return presets
+
+
+def config_version():
+    config_path = os.path.join(CONFIG_PATH, "config.json")
+    config = {}
+
+    if not os.path.exists(config_path):
+        return version.parse("0.4.0")
+
+    with open(config_path, "r") as file:
+        config = json.load(file)
+
+    if "version" in config.keys():
+        return version.parse(config["version"])
+
+    return version.parse("0.4.0")
+
+
+def _config_suffix():
+    deprecated_path = os.path.join(CONFIG_PATH, "config")
+    config_path = os.path.join(CONFIG_PATH, "config.json")
+    if os.path.exists(deprecated_path) and not os.path.exists(config_path):
+        logger.info('Moving "%s" to "%s"', deprecated_path, config_path)
+        os.rename(deprecated_path, config_path)
+
+
+def _preset_path():
+    """Migrate the folder structure from < 0.4.0.
+
+    Move existing presets into the new subfolder "presets"
+    """
+    new_preset_folder = os.path.join(CONFIG_PATH, "presets")
+    if os.path.exists(get_preset_path()) or not os.path.exists(CONFIG_PATH):
+        return
+
+    logger.info("Migrating presets from < 0.4.0...")
+    groups = os.listdir(CONFIG_PATH)
+    mkdir(get_preset_path())
+    for group in groups:
+        path = os.path.join(CONFIG_PATH, group)
+        if os.path.isdir(path):
+            target = path.replace(CONFIG_PATH, new_preset_folder)
+            logger.info('Moving "%s" to "%s"', path, target)
+            os.rename(path, target)
+
+    logger.info("done")
+
+
+def _mapping_keys():
+    if not os.path.exists(get_preset_path()):
+        return  # don't execute if there are no presets
+    for preset in all_presets():
+        preset_dict = {}
+        with open(preset, "r") as file:
+            preset_dict = json.load(file)
+        if "mapping" in preset_dict.keys():
+            mapping = copy.deepcopy(preset_dict["mapping"])
+            for key in mapping.keys():
+                if key.count(",") == 1:
+                    preset_dict["mapping"][f"{key},1"] = preset_dict["mapping"].pop(key)
+
+        with open(preset, "w") as file:
+            json.dump(preset_dict, file, indent=4)
+            file.write("\n")
+
+
+def _update_version():
+    config_file = os.path.join(CONFIG_PATH, "config.json")
+    if not os.path.exists(config_file):
+        return
+
+    logger.info("version in config file to %s", VERSION)
+    with open(config_file, "r") as file:
+        config = json.load(file)
+
+    config["version"] = VERSION
+    with open(config_file, "w") as file:
+        json.dump(config, file, indent=4)
+
+
+def migrate():
+    v = config_version()
+    if v < version.parse("1.2.1"):
+        _config_suffix()
+        _preset_path()
+        _mapping_keys()
+        _update_version()
```

Comment:
the other way around: First migrate to 0.4.0 

Since this is probably going to be implemented in 1.2.3 we should check `if v < 1.2.3` but this would make it difficult to test as long as we are on 1.2.2 so I suggest to introduce a 1.2.3.dev1 version with this PR